### ISSUE TITLE
    raise TypeError(repr(o) + " is not JSON seriali

### ISSUE THREAD

**Issue by DKorytkin** [DKorytkin/Checker#20]
example code

```python
Checker(Or({'test1': int}, {'key1': str})).validate({'test1': 'test'})
```

```python
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/IPython/core/interactiveshell.py", line 2882, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-7-7c4d74a94717>", line 1, in <module>
    Checker(Or({'test1': int}, {'key1': str})).validate({'test1': 'test'})
  File "/home/dkorytkin/json_checker/checker.py", line 325, in validate
    result = checker.validate(data)
  File "/home/dkorytkin/json_checker/checker.py", line 268, in validate
    result = self.expected_data.validate(data)
  File "/home/dkorytkin/json_checker/checker.py", line 205, in validate
    return self._format_errors(errors)
  File "/home/dkorytkin/json_checker/checker.py", line 195, in _format_errors
    self._format_data(),
  File "/home/dkorytkin/json_checker/checker.py", line 190, in _format_data
    return tuple(_format_data(d) for d in self.expected_data)
  File "/home/dkorytkin/json_checker/checker.py", line 190, in <genexpr>
    return tuple(_format_data(d) for d in self.expected_data)
  File "/home/dkorytkin/json_checker/checker.py", line 65, in _format_data
    return json.dumps(data)
  File "/usr/lib/python2.7/json/__init__.py", line 244, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON seriali

```

**Comment by DKorytkin** [comment]
```python
from checker import Checker, Or, And, OptionalKey

ex = [{'caption': basestring,
 'data': Or(
     {'max': Or(int, None), 'min': Or(int, None), OptionalKey('presets'): [int]}, 
     {'values': [{'count': int, 'caption': basestring, 'id': Or(int, basestring)}]}),
 'key': basestring}]

cu = [
    {'caption': 'price', 'data': {'max': 'one', 'min': None}, 'key': 'price'},
    {'caption': 'state', 'data': {'values': [{'caption': 'ru', 'count': 23, 'id': 1123},
    {'caption': 'fr', 'count': 21, 'id': 666}]},
  'key': 'bss1'}]

Checker(ex, soft=True).validate(cu)
```

**Comment by DKorytkin** [comment]
```python
json.dumps({'max': Or(int, None), OptionalKey('presets'): [int], 'min': Or(int, None)})

```

**Comment by DKorytkin** [comment]
implemented in #23
